src/publish/confluence/confluence-helper.ts

```diff
@@ -420,32 +508,66 @@ export const updateLinks = (
     parent.space
   }/pages/`;
 
-  const replacer = (match: string): string => {
-    let updated: string = match;
-    const fileNameMatch = FILE_FINDER.exec(match);
-    const fileName = fileNameMatch ? fileNameMatch[0] ?? "" : "";
+  const changeMapper = (
+    changeToProcess: ConfluenceSpaceChange
+  ): ConfluenceSpaceChange => {
+    const replacer = (match: string): string => {
+      let documentFileName = "";
+      if (
+        isContentUpdate(changeToProcess) ||
+        isContentCreate(changeToProcess)
+      ) {
+        documentFileName = changeToProcess.fileName ?? "";
+      }
 
-    const fileNameExtension = `${fileName}.qmd`;
+      const docFileNamePathList = documentFileName.split("/");
 
-    const sitePage: SitePage | null =
-      fileMetadataTable[fileNameExtension] ?? null;
-    if (sitePage) {
-      updated = match.replace('href="', `href="${url}`);
-      const pagePath: string = `${url}${sitePage.id}/${encodeURI(
-        sitePage.title ?? ""
-      )}`;
+      let updated: string = match;
+      const linkFileNameMatch = FILE_FINDER.exec(match);
 
-      updated = updated.replace(fileNameExtension, pagePath);
-    }
+      const linkFileName = linkFileNameMatch ? linkFileNameMatch[0] ?? "" : "";
 
-    return updated;
-  };
+      const fileNamePathList = linkFileName.split("/");
+
+      const linkFullFileName = `${linkFileName}.qmd`;
+
+      let siteFilePath = linkFullFileName;
+      const isAbsolute = siteFilePath.startsWith("/");
+      if (!isAbsolute && docFileNamePathList.length > 1) {
+        const relativePath = docFileNamePathList
+          .slice(0, docFileNamePathList.length - 1)
+          .join("/");
+
+        if (siteFilePath.startsWith("./")) {
```

Comment:
Just FYI, There is a path `join` function you can use to join paths that don't include platform specific separators (perhaps this is a link so doesn't matter).

Reply:
Thank you, I think I need to make that change.

Reply:
I need to research Deno support for resolving platform specific paths using the link address.  This is for both links and images.  I've created this to test and fix.  https://github.com/rstudio/quarto-confluence-test/issues/36